src/mono/mono/metadata/marshal.c

```diff
@@ -4183,6 +4185,55 @@ marshal_get_managed_wrapper (MonoMethod *method, MonoClass *delegate_klass, Mono
 	csig->hasthis = 0;
 	csig->pinvoke = 1;
 
+	if (invoke)
+		mono_marshal_set_callconv_from_modopt (invoke, csig, TRUE);
+	else
+		mono_marshal_set_callconv_from_unmanaged_callers_only_attribute(method, csig);
+
+	if (mono_method_signature_has_ext_callconv (csig, MONO_EXT_CALLCONV_SWIFTCALL)) {
+		MonoClass *swift_self = mono_class_try_get_swift_self_class ();
+		MonoClass *swift_error = mono_class_try_get_swift_error_class ();
+		MonoClass *swift_indirect_result = mono_class_try_get_swift_indirect_result_class ();
+		swift_lowering = g_newa (SwiftPhysicalLowering, sig->param_count);
```

Comment:
Where are these deallocated?

Reply:
Good catch. Added deallocation for `new_params`. `swift_lowering` and `swift_sig_to_csig_mp` are allocated on stack, so we don't deallocate explicetely them right?

Reply:
Yes. Why they are allocated differently?

Reply:
`swift_lowering` and `swift_sig_to_csig_mp` are going to have exactly `sig->param_count` elements. In some places across marshalling code such arrays are allocated on stack. 
 https://github.com/dotnet/runtime/blob/f9eda079502aa737ed7c75cf14dea10223cfb0fa/src/mono/mono/metadata/marshal-lightweight.c#L2797
 
 `swift_sig_to_csig_mp` is an array of `ints` so I guess it ok. We could make the `swift_lowering` heap allocated, but I don't know whether this is necessary.

The `new_params` can be up to 4 times longer than the`swift_lowering` and `swift_sig_to_csig_mp`, as it includes the lowered params. So I following what was done in method-to-ir. 

https://github.com/dotnet/runtime/blob/f9eda079502aa737ed7c75cf14dea10223cfb0fa/src/mono/mono/mini/method-to-ir.c#L7533